src/Symfony/Component/HttpFoundation/Session/Session.php

```diff
@@ -253,7 +253,9 @@ public function registerBag(SessionBagInterface $bag)
      */
     public function getBag($name)
     {
-        return $this->storage->getBag($name)->getBag();
+        $bag = $this->storage->getBag($name);
+
+        return method_exists($bag, 'getBag') ? $bag->getBag() : $bag;
```

Comment:
Why not test against `SessionBagProxy` to make the intention clear?

Reply:
Agree, otherwise i'd prefer to deprecate the situation (calling getBag on a non-proxy bag).

But i tend to believe one relies on a bug in this case.

Reply:
I don't know how many people out there created the own `SessionBagInterface` implementations and also implemented this method this method to work around this issue. IMO we should play safe here to not risk any existing application.